### DISCUSSION TITLE
Tidak bisa apply prediksi ke data test? (Capstone ML)

### DISCUSSION THREAD

**Discussion by AGelfP** [teamalgoritma/community#637] [Q&A] [answered]
Dear Team Algoritma,

Ini saya udah mengikuti petunjuk yang diberikan di halaman rpubs untuk memprediksi data test tapi kok malah dianggap error ya? Berikut objek-objek saya:

```
# ini model dengan MAE terendah
sarima_fnb <- forecast(mod_msarima, h = 13*7) 

# ini pembacaan data testnya
data_test <- read.csv("data/data-test.csv")
submission <- data_test %>% 
  mutate(datetime = ymd_hms(datetime))

# ini aplikasi modelnya
 submission %>% 
  mutate(visitor = sarima_fnb)
```

Berikut adalah pesan errornya

```
Error in `mutate()`:
ℹ In argument: `visitor = sarima_fnb`.
Caused by error:
! `visitor` must be a vector, not a <adam.forecast/smooth.forecast/forecast> object.
Backtrace:
 1. submission %>% mutate(visitor = sarima_fnb)
 9. dplyr:::dplyr_internal_error("dplyr:::mutate_not_vector", `<named list>`)
Error in mutate(., visitor = sarima_fnb) :
Caused by error:
! `visitor` must be a vector, not a <adam.forecast/smooth.forecast/forecast> object.
```

Apakah ada hal yang perlu saya tambahkan sebelum saya aplikasikan modelnya, atau apakah ada kesalahan lain ya?  Mohon bantuannya. Terima kasih.

**Comment by amaliasf** [comment, marked as answer]
Dear Pak @AGelfP 

Terimakasih untuk pertanyaanya.

Terkait pesan error yang Bapak terima sepertinya dikarenakan object `sarima_fnb` bukan merupakan suatu nilai vector. 

Untuk melakukan subtitusi nilai `visitor` dengan hasil prediksi, maka kita hanya butuh suatu nilai vector yang menampung nilai dari hasil prediksi yang diperoleh.

Maka untuk bisa mengambil nilai titik prediksi dari object `sarima_fnb`, kita dapat menggunakan `sarima_fnb$mean`. 

Demikian yang dapat kami sampaikan, semoga membantu.
